lib/mongo/protocol.ex

```diff
@@ -125,7 +131,17 @@ defmodule Mongo.Protocol do
   end
 
   def handle_execute(%Mongo.Query{action: action, extra: extra}, params, opts, s) do
-    handle_execute(action, extra, params, opts, s)
+    :ok = :inet.setopts(s.socket, [active: false])
+    with {:ok, reply, new_s} <- handle_execute(action, extra, params, opts, %{
+           s | database: Keyword.get(opts, :database, s.database)
+         }) do
```

Comment:
Break the state change out into a variable, this is hard to read.

Do we need separation of `new_s` and `s`?